cpp/include/xtt/cpp.hpp

```diff
@@ -0,0 +1,35 @@
+/******************************************************************************
+ *
+ * Copyright 2018 Xaptum, Inc.
+ * 
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ * 
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+
+#ifndef XTT_CPP_XTTCPP_HPP
```

Comment:
I'm now thinking this file could simply be `include/xtt.hpp` and the included headers just `include/xtt/identity.hpp`, e.g.  The `.h` vs `.hpp` endings prevents collisions between the C and C++ libs.

`#include <xtt.hpp>` seems more correct than `#include <xtt/cpp.hpp>`.

(`xtt-asio` would still be `#include <xtt/asio.hpp>)